#! /usr/bin/env python
# MPEPC Planner configuration

PACKAGE='mpepc_local_planner'

import sys

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
# Trajectory Model Params
gen.add("cl_K_1", double_t, 0,
    "Control law K1", 1.2, 0.0, 10.0)
gen.add("cl_K_2", double_t, 0,
    "Control law K2", 3.0, 0.0, 10.0)
gen.add("cl_BETA", double_t, 0,
    "Control law Beta", 0.4, 0.0, 1.0)
gen.add("cl_LAMBDA", double_t, 0,
    "Control law Lambda", 2.0, 0.0, 10.0)
gen.add("cl_R_THRESH", double_t, 0,
    "R Threshold", 0.05, 0.0, 1.0)
gen.add("cl_V_MAX", double_t, 0,
    "Velocity max", 0.5, 0.0, 1.5)
gen.add("cl_V_MIN", double_t, 0,
    "Velocity min", 0.0, 0.0, 1.0)
gen.add("cl_W_TURN", double_t, 0,
    "Angular velocity turn to goal", 0.2, 0.0, 1.0)

# Trajectory Optimization Params
gen.add("TIME_HORIZON", double_t, 0,
    "MPEPC Horizon", 4.0, 0.0, 10.0)
gen.add("DELTA_SIM_TIME", double_t, 0,
    "Delta simulate time", 0.2, 0.0, 1.0)
gen.add("SAFETY_ZONE", double_t, 0,
    "Minimum distance to obstacle", 0.225, 0.0, 1.0)
gen.add("WAYPOINT_THRESH", double_t, 0,
    "Waypoint threshold", 1.75, 0.0, 5.0)
    

exit(gen.generate(PACKAGE, "mpepc_local_planner", "MPEPCPlanner"))
